### ISSUE TITLE
Sphinx v1.8.0 complain that "extension 'recommonmark' has no setup() function"

### ISSUE THREAD

**Issue by leemeans** [rtfd/recommonmark#119]
I'm using recommonmark with the lastest version of Sphinx, and i got the warning below :
```
sphinx-build -M text policy policy/_build
Running Sphinx v1.8.0
WARNING: extension 'recommonmark' has no setup() function; is it really a Sphinx extension module?
WARNING: Unknown configure key: latex_elements['bable']. ignored.
```

How could i fix it or ignore it? Thanks.

**Comment by stsewd** [comment]
How are you installing the extension? Can I see your conf.py file? 

We have a setup function https://github.com/rtfd/recommonmark/blob/33b5c2a4ec50d18d3f659aa119d3bd11452327da/recommonmark/__init__.py#L6

**Comment by leemeans** [comment]
This is my conf.py file screenshot:

![sphinx](https://user-images.githubusercontent.com/15212302/45669038-4ccc4280-bb51-11e8-9cae-6e50c2e4a07c.png)

Well, my Sphinx  version is 1.8.0 , so i add the recommmonmark as suggested [here](https://github.com/rtfd/recommonmark/blob/master/README.md#getting-started)

**Comment by stsewd** [comment]
Are you installing recommonmark from pip?

**Comment by leemeans** [comment]
Yes, i use `sudo pip install recommonmark` to install it.And now the requirement is satisfied:
```
 leemeans  ~  sudo pip install recommonmark 
Requirement already satisfied: recommonmark in /usr/local/lib/python2.7/dist-packages (0.4.0)
Requirement already satisfied: docutils>=0.11 in /usr/lib/python2.7/dist-packages (from recommonmark) (0.14)
Requirement already satisfied: commonmark<=0.5.4 in /usr/local/lib/python2.7/dist-packages (from recommonmark) (0.5.4)
```
import from python prompt works:
```
 leemeans  ~  sudo pip install recommonmark 
Requirement already satisfied: recommonmark in /usr/local/lib/python2.7/dist-packages (0.4.0)
Requirement already satisfied: docutils>=0.11 in /usr/lib/python2.7/dist-packages (from recommonmark) (0.14)
Requirement already satisfied: commonmark<=0.5.4 in /usr/local/lib/python2.7/dist-packages (from recommonmark) (0.5.4)
```

So, as far as i can see, it's so weird.

**Comment by Antoinelypro** [comment]
I got the same issue.

Below is my conf.py

```

# -- General configuration ---------------------------------------------------

# If your documentation needs a minimal Sphinx version, state it here.
#
# needs_sphinx = '1.0'

# Add any Sphinx extension module names here, as strings. They can be
# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
# ones.
extensions = [
    'sphinx.ext.autodoc',
    'sphinx.ext.doctest',
    'sphinx.ext.intersphinx',
    'sphinx.ext.todo',
    'sphinx.ext.coverage',
    'sphinx.ext.mathjax',
    'sphinx.ext.ifconfig',
    'sphinx.ext.viewcode',
    'sphinx.ext.githubpages',
    'sphinx.ext.napoleon',
    'recommonmark'
]

# Add any paths that contain templates here, relative to this directory.
templates_path = ['_templates']

def setup(app):
    app.add_config_value('recommonmark_config', {
            'url_resolver': lambda url: github_doc_root + url,
            'auto_toc_tree_section': 'Contents',
            }, True)
    app.add_transform(AutoStructify)

source_parsers = {
   '.md': 'recommonmark.parser.CommonMarkParser',
}

# The suffix(es) of source filenames.
# You can specify multiple suffix as a list of string:
#
source_suffix = ['.rst', '.md']
#source_suffix = '.rst'



# The master toctree document.
master_doc = 'index'

# The language for content autogenerated by Sphinx. Refer to documentation
# for a list of supported languages.
#
# This is also used if you do content translation via gettext catalogs.
# Usually you set "language" from the command line for these cases.
language = None

# List of patterns, relative to source directory, that match files and
# directories to ignore when looking for source files.
# This pattern also affects html_static_path and html_extra_path.
exclude_patterns = []

# The name of the Pygments (syntax highlighting) style to use.
pygments_style = None


# -- Options for HTML output -------------------------------------------------

# The theme to use for HTML and HTML Help pages.  See the documentation for
# a list of builtin themes.
#
html_theme = 'nature'

# Theme options are theme-specific and customize the look and feel of a theme
# further.  For a list of options available for each theme, see the
# documentation.
#
# html_theme_options = {}

# Add any paths that contain custom static files (such as style sheets) here,
# relative to this directory. They are copied after the builtin static files,
# so a file named "default.css" will overwrite the builtin "default.css".
html_static_path = ['_static']

# Custom sidebar templates, must be a dictionary that maps document names
# to template names.
#
# The default sidebars (for documents that don't match any pattern) are
# defined by theme itself.  Builtin themes are using these templates by
# default: ``['localtoc.html', 'relations.html', 'sourcelink.html',
# 'searchbox.html']``.
#
# html_sidebars = {}


# -- Options for HTMLHelp output ---------------------------------------------

# Output file base name for HTML help builder.
htmlhelp_basename = 'err_estimationdoc'


# -- Options for LaTeX output ------------------------------------------------

latex_elements = {
    # The paper size ('letterpaper' or 'a4paper').
    #
    # 'papersize': 'letterpaper',

    # The font size ('10pt', '11pt' or '12pt').
    #
    # 'pointsize': '10pt',

    # Additional stuff for the LaTeX preamble.
    #
    # 'preamble': '',

    # Latex figure (float) alignment
    #
    # 'figure_align': 'htbp',
}

# Grouping the document tree into LaTeX files. List of tuples
# (source start file, target name, title,
#  author, documentclass [howto, manual, or own class]).
latex_documents = [
    (master_doc, 'err_estimation.tex', 'err\\_estimation Documentation',
     'Antoine Ly', 'manual'),
]


# -- Options for manual page output ------------------------------------------

# One entry per manual page. List of tuples
# (source start file, name, description, authors, manual section).
man_pages = [
    (master_doc, 'err_estimation', 'err_estimation Documentation',
     [author], 1)
]


# -- Options for Texinfo output ----------------------------------------------

# Grouping the document tree into Texinfo files. List of tuples
# (source start file, target name, title, author,
#  dir menu entry, description, category)
texinfo_documents = [
    (master_doc, 'err_estimation', 'err_estimation Documentation',
     author, 'err_estimation', 'One line description of project.',
     'Miscellaneous'),
]


# -- Options for Epub output -------------------------------------------------

# Bibliographic Dublin Core info.
epub_title = project

# The unique identifier of the text. This can be a ISBN number
# or the project homepage.
#
# epub_identifier = ''

# A unique identification for the text.
#
# epub_uid = ''

# A list of files that should not be packed into the epub file.
epub_exclude_files = ['search.html']


# -- Extension configuration -------------------------------------------------

# -- Options for intersphinx extension ---------------------------------------

# Example configuration for intersphinx: refer to the Python standard library.
intersphinx_mapping = {'https://docs.python.org/': None}

# -- Options for todo extension ----------------------------------------------

# If true, `todo` and `todoList` produce output, else they produce nothing.
todo_include_todos = True



```

**Comment by cas--** [comment]
This is how to enable markdown in Sphinx: http://www.sphinx-doc.org/en/master/usage/markdown.html

The recommonmark docs should be updated...

**Comment by townsendbt** [comment]
Above suggestion didn't work for me. I'm still getting the warning using Sphinx 1.8

**Comment by cas--** [comment]
@townsendbt Please provide your conf.py

**Comment by townsendbt** [comment]
sorry @cas-- I didn't see your update.
Issue is resolved, thanks anyway.
I had used pip to install. I uninstalled, cloned the repo from GH and installed from there. I had v 0.4.0 from pip.

**Comment by ericholscher** [comment]
I believe this should be fixed in 0.5.0 which is now released.
